### DISCUSSION TITLE
[GSoC admin] Contributor selection and ranking criteria for GSoC 2023

### DISCUSSION THREAD

**Discussion by aliok** [cncf/mentoring#860] [Google Summer of Code 2023]
I offered my help to manage GSoC 2023 program at CNCF and I am curious about the contributor selection and ranking criteria at CNCF.  I wasn't doing it last year.

I would be happy to document the process described here into a document. (and maybe do even more stuff for https://github.com/cncf/mentoring/issues/201)

**Comment by nate-double-u** [comment]
Last year I put together a shortlist in a google sheet, then had a conversation with @caniszczyk to confirm our selections.

This year, I expect the shortlist and confirmation will be done by the folks in the Mentorship WG who are on the GSoC administration team (currently myself, @aliok, and @jaytiaki).

**Comment by aliok** [comment]
I think it would be useful we we can specify things like that for CNCF. Google lets orgs implement their own selection criteria.

Let me start with what we had at JBoss community (I was a GSoC admin there for 4 years):
- We left the main acceptance to the mentors with these caveats:
  - Contributor has to have a meaningful contribution (no typo fix only) already: Although this might seem to have a conflict with Google's definition of "open source beginner", what we expected were not big feature implementations or something like that. Instead,  we wanted something that shows the applicant is capable and understands the problem. We accepted applicants who were at least submitting a bug fix, or finding issues with docs/tutorials, or preparing a demo using the project. Definition of this criteria is hard to write in very specific details but we tried to give a good message to our mentors around that.
  - Contributor has to have a good proposal, showing that they have a good understanding of the problem.
  - Contributor has to have a good community engagement. We didn't prefer contributors reaching out to mentors privately only.
- We ranked proposals with more than 1 mentors higher
- To give fair chance to communities, we didn't prefer having multiple projects for communities (depends on mentor count of course)
- **Most importantly**, we advised mentors to only press the "Want to Mentor" button for applicants who they will really want to **professionally** work with for 3 months, 3-4 hours a week.

What was not in criteria:
- We never asked if applicants are willing to stay active after GSoC and do the selection based on that
- We never asked the motivation for applicants; it is their business


> Last year I put together a shortlist in a google sheet, then had a conversation with @caniszczyk to confirm our selections. 
> This year, I expect the shortlist and confirmation will be done by the folks in the Mentorship WG who are on the GSoC administration team (currently myself, @aliok, and @jaytiaki).

It is actually useful to talk about the process as well; thanks for bringing it up. I think having a Google sheet and a conversation would work fine :)

**Comment by aliok** [comment]
Created https://github.com/cncf/mentoring/pull/1005 out of this discussuion
